add scripts and workflow to build on GitHub Actions

kramred opened the pull request:
This is a workflow for Github Actions that will build macOS binaries for Ungoogled Chromium in 2-3 sequential steps and then upload them as artifact.

It is triggered by pushing to any `release/*` branch.

Build times are in the range of 11 to 13 hours.

See [here](https://github.com/kramred/ungoogled-chromium-macos/actions/runs/169137335) for an example of a finished run.

I hope by this people can better trust the binaries.

23rd commented on github_get_env.sh:
```diff
@@ -0,0 +1,14 @@
+sudo kextstat
+df -h && ls -kahl
+sudo mdutil -a -i off # disable spotlight
+sudo mdutil -d /
```
Nitpick: would be better to add an empty line to EOF.

kramred replied:
I was wondering why to add a `newline` to the last line and found [this of stackoverflow](https://stackoverflow.com/a/729795) referring to the [POSIX.1-2017 definitions of a line](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206) – if anyone else is wondering too.
Will change the files to include a final `newline` character

23rd commented on .github/workflows/build.yml:
```diff
@@ -1,38 +1,119 @@
+        uses: actions/upload-artifact@v2
+        with:
+          name: github_build_logs
+          path: upload_logs/
```
Nitpick: would be better to add an empty line to EOF.

23rd commented:
Perhaps it would be better to move all `github_*.sh` files to the `.github/workflows/` directory?

kramred commented:
> Perhaps it would be better to move all `github_*.sh` files to the `.github/workflows/` directory?

I have no strong opinion on that. They could be moved to the `.github/workflows/` directory, so that everything concerning the github actions build is in one place. On the other hand they might be useful for people wanting to modify their local build scripts and would be more visible in the main directory.

Other arguments on where to keep these files? Or the repo maintainer may decide…

anshul78 commented:
> Perhaps it would be better to move all github_*.sh files to the .github/workflows/ directory?

Generally, `.github/workflows/` is used only for workflow files. So another directory, say, `.github/scripts/` can be used to place them.

Eloston commented:
I agree with @anshul78, I think they should be placed in `.github/scripts`. I don't see a use-case where our typical user would want to run the GitHub workflow scripts on their machine instead of running the build scripts already in this repo.

Also, why would you trigger the build on any `release/*` branch? We don't have any branch scheme like that right now, so what are you proposing?

kramred commented:
> I agree with @anshul78, I think they should be placed in `.github/scripts`. I don't see a use-case where our typical user would want to run the GitHub workflow scripts on their machine instead of running the build scripts already in this repo.

I've moved the github specific scripts to `.github/scripts`
 
> Also, why would you trigger the build on any `release/*` branch? We don't have any branch scheme like that right now, so what are you proposing?

The idea here was to prevent running a full build on changes to the documentation, i.e. anything not build related.
I've changed the triggers in the yaml file to specify paths and files that should trigger a build, or ones that should be ignored.

kramred commented:
I've just converted this PR to draft status. I want to include a step to publish a release with the generated diskimage. I will update status when this is working.

anshul78 commented:
I don't see why the workflow needs to be run on pull request. If it is run, it should only check whether the binary gets compiled.

For commits, it should run on all push (which should include merged pull request?) except changes to specific files/folders.

kramred commented:
This has finally finished building and creating a release is working too, see [here](https://github.com/kramred/ungoogled-chromium-macos/releases/tag/84.0.4147.89-1.2) – opening again for review
The idea would then be to directly link to the latest GitHub release on the binaries web-site
